devcards.js

```diff
@@ -5,42 +5,56 @@ var $ = React.createElement;
 
 var DevCards = require('./lib/components/DevCards');
 
-exports = module.exports = devcard;
-exports.run = run;
-
-/**
- * Cards
- */
-
-// Main card
-function devcard(name, doc, body) {
-  devcard.current.push({name: name, doc: doc, body: body});
+var DEVCARDS_DIV_ID = '__devcards-root';
+
+// Singleton state
+// TODO: eliminate?
+var catalog = {};
+
+exports.ns = ns;
+function ns(name) {
+  // Because execution of a loaded module is synchronous
+  // we can expect a reloading namespace to be filled
+  // in a single pass of the event loop
+  // therefore we can enqueue a reRender for the next tick
+  var cards = catalog[name] = [];
+  function add(card) {
+    cards.push(card);
+    enqueueRender();
+  }
+  return createDefcard(add);
 }
 
-// Noop card
-devcard.off = Function.prototype;
+function createDefcard(add) {
+  // Main card
+  function defcard(name, doc, body) {
+    add({name: name, doc: doc, body: body});
+  }
+
+  // Noop card
+  defcard.off = Function.prototype;
 
+  return defcard;
+}
 
 /**
- * Runtime
+ * Rendering
  */
-
-// Rendering the cards
-function render(catalog, rootId) {
+function getRoot() {
   /* eslint-env browser */
-  var root = document.getElementById(rootId);
-  ReactDOM.render($(DevCards, { catalog: catalog }), root);
-}
-
-// Running the cards
-function run(modules, require, opts) {
-  opts = opts || {};
+  var root = document.getElementById(DEVCARDS_DIV_ID);
+  if (root) return root;
 
-  var catalog = {};
-  modules.forEach(function(m) {
-    devcard.current = catalog[m] = [];
-    require(m);
-  });
+  var newRoot = document.createElement('div');
+  newRoot.id = DEVCARDS_DIV_ID;
+  document.body.appendChild(newRoot);
+  return newRoot;
+}
 
-  render(catalog, opts.rootId || 'DEVCARDS');
+function enqueueRender() {
+  clearTimeout(enqueueRender.timer);
+  enqueueRender.timer = setTimeout(render, 0);
+}
+function render() {
```

Comment:
The flash of the logo is provided by the equivalent of the hot-middleware client (figwheel), rather than devcards